validator/client/validator.go

```diff
@@ -369,6 +370,85 @@ func (v *validator) SlotDeadline(slot types.Slot) time.Time {
 	return time.Unix(int64(v.genesisTime), 0 /*ns*/).Add(secs * time.Second)
 }
 
+// CheckDoppelGanger checks if the current actively provided keys have
+// any duplicates active in the network.
+func (v *validator) CheckDoppelGanger(ctx context.Context) error {
+	if !featureconfig.Get().EnableDoppleGanger {
+		return nil
+	}
+	log.Info("Running Doppleganger Check")
+	pubkeys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
Optional, but i think this is a nicer UX. I moved the log after the fetching validating keys so we could log the number of keys. Also removed some extra capitalization.

```suggestion
	pubkeys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
	if err != nil {
		return err
	}
	log.WithField("keys", len(pubkeys)).Info("Running doppleganger check")
```